http-ballerina/http_errors.bal

```diff
@@ -14,152 +14,156 @@
 // specific language governing permissions and limitations
 // under the License.
 
-# Holds the details of an HTTP error
+# Represents the details of an HTTP error.
 # 
 # + statusCode - The status code, if the inbound error response exists
 public type Detail record {
     int statusCode?;
 };
 
-// Ballerina HTTP Client Error Types
+# Represents the details of the `LoadBalanceActionError`.
+#
+# + httpActionErr - Array of errors occurred at each endpoint
+public type LoadBalanceActionErrorData record {
+    error[] httpActionErr?;
+};
+
+// Level 1
+# Defines the common error type for the module
+public type Error distinct error;
+
+// Level 2
+# Defines the possible listener error types
+public type ListenerError distinct Error;
```

Comment:
Do we need separate `ListenerError` and `ClientError`?

Reply:
Thought it would create a clear separation between client and listener errors. 
https://docs.google.com/spreadsheets/d/1bkOn7TahMyz22HC1XlE8x7-ByIZV1AZf6eDC4sRXEXM/edit?usp=sharing

Reply:
In a recent discussion, we agreed to not to have a separation between client and listener errors, since they cannot occur at the same time. (Not sure if that applies to the HTTP scenario though).

Reply:
Yeah, I was thinking the same. IMO, no point in having both even in HTTP.

Reply:
From a single method POV, there is no such occurrence where both client and listener errors are returned. But there could be situations such as `trap` and `do {} on fail{}` where you capture both types. In such cases, this separation help user to isolate the root cause up to a certain extent. 

I have introduces common error `http:Error` also, so the user has the option to use it when the separation is not needed.